executor/src/lib.rs

```diff
@@ -522,11 +534,18 @@ where
             let _timer = OP_COUNTERS.timer("storage_save_transactions_time_s");
             OP_COUNTERS.observe("storage_save_transactions.count", num_txns_to_commit as f64);
             assert_eq!(first_version_to_commit, version + 1 - num_txns_to_commit);
-            self.storage_write_client.save_transactions(
-                txns_to_commit,
-                first_version_to_commit,
-                Some(ledger_info_with_sigs.clone()),
-            )?;
+            let write_client = self.storage_write_client.clone();
+            let ledger_info_with_sigs = ledger_info_with_sigs.clone();
+            block_on(self.rt.spawn(async move {
+                write_client
+                    .save_transactions_async(
+                        txns_to_commit,
+                        first_version_to_commit,
+                        Some(ledger_info_with_sigs),
+                    )
+                    .await
+            }))
+            .unwrap()?;
```

Comment:
Is this unwrapping the spawn result?

Reply:
in 0.2 when you spawn a future you get back a JoinHandle which resolves to a the output of the future you spawned, or rather `Result<OutputOfFuture, FuturePanic'edError>` The only time a JoinHandle returns an error is if the future panic'ed at which point there's nothing really we can do to recover.